### PR TITLE
Worker robustness improvements

### LINKED ISSUE #1006: Outdated method docstring in docker image manager
https://github.com/codalab/codalab-cli/blob/e76863278421a6b91af86f7ac968e36f727ba4b7/worker/codalabworker/local_run/docker_image_manager.py#L123

### LINKED ISSUE #1154: Dependency symlink [File exists] issue in worker
Many workers went offline on the same day with the following error message:
```
2019-06-03 19:53:31,621 http://localhost:None "GET /v1.35/images/theerit/quac_flowqa_submission:latest/json HTTP/1.1" 200 None
Traceback (most recent call last):
  File "/opt/worker/codalabworker/worker.py", line 51, in start
    self._run_manager.process_runs()
  File "/opt/worker/codalabworker/local_run/local_run_manager.py", line 171, in process_runs
    self._runs[bundle_uuid] = self._run_state_manager.transition(run_state)
  File "/opt/worker/codalabworker/fsm.py", line 35, in transition
    return self._transition_functions[state.stage](state)
  File "/opt/worker/codalabworker/local_run/local_run_state.py", line 221, in _transition_from_PREPARING
    os.symlink(docker_dependency_path, child_path)
OSError: [Errno 17] File exists

```

Unclear as of yet why this could be happening.

### PR DESCRIPTION
**Pull request by bkgoksel**
1. Make sure dependency child paths are nonempty 08778cb : should fix #1154 
2. Fix docker image manager docstring method 6b77c79 : should fix #1006 
3. More robust exception handling in worker (catch any unhandled exception for each run and only fail the run instead of crashing) 2959473
4. Connect to the correct local instance when launching tests from the service script ad43130 & 11a53da

### CODE REVIEW

File: worker/codalabworker/local_run/local_run_state.py [hunk 1]
```diff
@@ -106,13 +106,31 @@ def __init__(
...
+        )
+        self.add_transition(
+            LocalRunStage.UPLOADING_RESULTS,
+            self._transition_from_UPLOADING_RESULTS,
+            exception_handler_function=self._handle_exception,
+        )
+        self.add_transition(
+            LocalRunStage.FINALIZING,
+            self._transition_from_FINALIZING,
+            exception_handler_function=self._handle_exception,
```
Comment (percyliang) [inline worker/codalabworker/local_run/local_run_state.py:132]: If every stage has the same handler, why is it an argument?

**Review by percyliang — APPROVED** [review]

File: worker/codalabworker/local_run/local_run_state.py (hunk 1 above)
Comment (bkgoksel) [reply worker/codalabworker/local_run/local_run_state.py:132]: Yeah I was undecided between making it one handler for all exceptions versus specifying individual, and initially I thought I might need different behavior for each (which I kinda do now as well, if you look at the if statement inside the handler function). 

I should probably make it one default for all states, just to keep things simple.

**Comment by bkgoksel** [comment]
Enforcing dependency child paths to be nonempty breaks make bundles, which allow this. I need to update the logic there...

**Comment by bkgoksel** [comment]
Closing as split into PRs #1156 #1157 #1159

**Pull request closed without merging**